File: ios/Sources/KlinikAISettingsFeature/AISettingsScreen.swift
import SwiftUI
import KlinikAPI
import KlinikCore
import KlinikDesign

public enum AISettingsPhase: Sendable, Equatable {
    case loading
    case loaded
    case notPermitted
    case failed(String)
}

public struct AISettingsState: Sendable, Equatable {
    public var phase: AISettingsPhase = .loading
    public var settings: AISettings?
    public var providers: [AIProviderInfo] = []
    public var testResult: AIConnectionTest?
    public var saving = false
    public var error: String?

    public init() {}

    public func provider(_ choice: AIProviderChoice?) -> AIProviderInfo? {
        providers.first { $0.id == choice }
    }
}

/**
 * Which model the clinic uses, and on what terms (spec §3.4).
 *
 * Two gates, and the screen exists mostly to keep them visible. A provider with
 * a key and a model is *ready*; a provider whose zero-retention terms somebody
 * has confirmed is ready for **clinical** work, and the layer refuses clinical
 * prompts without the second. A screen that showed only the first would tell a
 * clinic it was finished while the half that matters stayed switched off.
 *
 * Prices are typed in, not fetched. A price this app carried would go stale in
 * a quarter and stay authoritative-looking while it did.
 */
@MainActor
public final class AISettingsModel {
    private let api: AISettingsAPI
    private var state = AISettingsState()

    public init(api: AISettingsAPI) {
        self.api = api
    }

    public func currentState() -> AISettingsState { state }

    public func load() async {
        do {
            async let settings = api.settings()
            async let providers = api.providers()

            state.settings = try await settings
            state.providers = try await providers
            state.phase = .loaded
        } catch APIError.forbidden {
            state.phase = .notPermitted
        } catch let error as APIError {
            state.phase = .failed(L10n.message(for: error))
        } catch {
            state.phase = .failed(L10n.string("error.server"))
        }
    }

    public func save(
        provider: AIProviderChoice?,
        model: String?,
        apiKey: String?,
        inputPrice: String?,
        outputPrice: String?,
        budget: String?
    ) async -> Bool {
        state.saving = true
        state.error = nil

        defer { state.saving = false }

        do {
            // An empty key means "leave the stored one alone" — which is what
            // changing a price should do, and what a write-only field has to
            // mean if it is not to wipe a credential by omission.
            state.settings = try await api.update(
                provider: provider,
                model: model,
                apiKey: apiKey?.isEmpty == true ? nil : apiKey,
                inputPricePerMTok: inputPrice,
                outputPricePerMTok: outputPrice,
                monthlyBudgetUsd: budget
            )

            return true
        } catch let error as APIError {
            state.error = L10n.message(for: error)
        } catch {
            state.error = L10n.string("error.server")
        }

        return false
    }

    /// Confirming the provider's zero-retention terms. Recorded with a note,
    /// because "somebody ticked a box" is not a contract and the record should
    /// say who checked what.
    public func confirmRetention(_ confirmed: Bool, note: String) async {
        state.saving = true
        state.error = nil

        defer { state.saving = false }

        do {
            state.settings = try await api.update(
                zeroRetentionConfirmed: confirmed,
                zeroRetentionNote: note.isEmpty ? nil : note
            )
        } catch let error as APIError {
            state.error = L10n.message(for: error)
        } catch {
            state.error = L10n.string("error.server")
        }
    }

    public func test() async {
        state.saving = true
        state.testResult = nil

        defer { state.saving = false }

        state.testResult = try? await api.test()
    }

    public func switchOff() async {
        state.saving = true

        defer { state.saving = false }

        state.settings = try? await api.clear()
        state.testResult = nil
    }
}

/// The AI provider settings screen.
public struct AISettingsScreen: View {
    @Environment(\.colorScheme) private var scheme
    @Environment(\.openURL) private var openURL

    private let model: AISettingsModel

    @State private var state = AISettingsState()
    @State private var provider: AIProviderChoice?
    @State private var chosenModel = ""
    @State private var apiKey = ""
    @State private var inputPrice = ""
    @State private var outputPrice = ""
    @State private var budget = ""
    @State private var retentionNote = ""

    public init(model: AISettingsModel) {
        self.model = model
    }

    public var body: some View {
        ScrollView {
            VStack(alignment: .leading, spacing: Tokens.Spacing.lg) {
                ErrorBanner(message: state.error)

                switch state.phase {
                case .loading:
                    SkeletonCard(lines: 5)
                        .accessibilityElement()
                        .accessibilityLabel(L10n.string("common.loading"))

                case .notPermitted:
                    MessageState(icon: "lock", text: L10n.string("ai.notPermitted"))
                        .frame(minHeight: 240)

                case .failed(let message):
                    MessageState(
                        icon: Tokens.State.labCritical.iconName,
                        text: message,
                        retryTitle: L10n.string("common.retry")
                    ) {
                        await reload()
                    }

                case .loaded:
                    status
                    chooser
                    pricing
                    retention
                    actions
                }
            }
            .padding(Tokens.Spacing.lg)
        }
        .background(Tokens.Palette.background.resolve(for: scheme))
        .navigationTitle(L10n.string("ai.settings.title"))
        .task { await reload() }
    }

    /// The two gates, said apart.
    @ViewBuilder
    private var status: some View {
        if let settings = state.settings {
            Card(tone: settings.readyForClinicalUse ? .success : .warning) {
                VStack(alignment: .leading, spacing: Tokens.Spacing.sm) {
                    Label(
                        L10n.string(settings.ready ? "ai.ready" : "ai.notReady"),
                        systemImage: settings.ready ? "checkmark.circle" : "exclamationmark.circle"
                    )
                    .font(Tokens.Typography.subheadingRelative)
                    .foregroundStyle(
                        (settings.ready ? Tokens.Palette.success : Tokens.Palette.warning)
                            .resolve(for: scheme)
                    )

                    if settings.ready, !settings.zeroRetentionConfirmed {
                        Text(L10n.string("ai.settings.notClinicalReady"))
                            .font(Tokens.Typography.calloutRelative)
                            .foregroundStyle(Tokens.Palette.warning.resolve(for: scheme))
                            .fixedSize(horizontal: false, vertical: true)
                    }

                    ForEach(settings.localizedMissing, id: \.self) { missing in
                        Label(missing, systemImage: "circle")
                            .font(Tokens.Typography.captionRelative)
                            .foregroundStyle(Tokens.Palette.textSecondary.resolve(for: scheme))
                    }

                    if let test = state.testResult {
                        Text(
                            test.ok
                                ? L10n.string("ai.settings.testOk")
                                    .replacingOccurrences(of: "{model}", with: test.model ?? "")
                                : "\(L10n.string("ai.settings.testFailed")): \(test.error ?? "")"
                        )
                        .font(Tokens.Typography.captionRelative)
                        .foregroundStyle(
                            (test.ok ? Tokens.Palette.success : Tokens.Palette.critical)
                                .resolve(for: scheme)
                        )
                        .fixedSize(horizontal: false, vertical: true)
                    }
                }
            }
        }
    }

    private var chooser: some View {
        VStack(alignment: .leading, spacing: Tokens.Spacing.md) {
            SectionHeader(title: L10n.string("ai.provider"))

            ForEach(state.providers) { info in
                Button {
                    provider = info.id
                    chosenModel = info.models.first ?? ""
                } label: {
                    Card(tone: provider == info.id ? .info : .neutral) {
                        VStack(alignment: .leading, spacing: Tokens.Spacing.sm) {
                            HStack {
                                Text(info.label)
                                    .font(Tokens.Typography.subheadingRelative)
                                    .foregroundStyle(
                                        Tokens.Palette.textPrimary.resolve(for: scheme)
                                    )

                                Spacer(minLength: Tokens.Spacing.sm)

                                if provider == info.id {
                                    Image(systemName: "checkmark.circle.fill")
                                        .foregroundStyle(
                                            Tokens.Palette.accent.resolve(for: scheme)
                                        )
                                        .accessibilityHidden(true)
                                }
                            }

                            // The provider's own words about what it keeps.
                            // Rendered rather than summarised: the clinic is
                            // confirming these terms, not our reading of them.
                            Text(info.retentionNote)
                                .font(Tokens.Typography.captionRelative)
                                .foregroundStyle(
                                    Tokens.Palette.textSecondary.resolve(for: scheme)
                                )
                                .fixedSize(horizontal: false, vertical: true)

                            if let url = URL(string: info.pricingUrl) {
                                Button(L10n.string("ai.pricingPage")) { openURL(url) }
                                    .font(Tokens.Typography.captionRelative)
                                    .frame(minHeight: Tokens.minimumTouchTarget)
                                    .foregroundStyle(
                                        Tokens.Palette.accent.resolve(for: scheme)
                                    )
                            }
                        }
                    }
                    .contentShape(Rectangle())
                }
                .buttonStyle(.plain)
                .accessibilityElement(children: .combine)
                .accessibilityAddTraits(
                    provider == info.id ? [.isButton, .isSelected] : .isButton
                )
            }

            if let info = state.provider(provider), !info.models.isEmpty {
                Picker(L10n.string("ai.model"), selection: $chosenModel) {
                    ForEach(info.models, id: \.self) { name in
                        Text(name).tag(name)
                    }
                }
                .pickerStyle(.menu)
                .frame(minHeight: Tokens.minimumTouchTarget)
                .accessibilityLabel(L10n.string("ai.model"))
            }

            LabelledField(
                label: L10n.string("ai.settings.apiKey"),
                text: $apiKey,
                isSecure: true,
                contentType: .password,
                keyboard: .default
            )

            if let last4 = state.settings?.apiKeyLast4 {
                Text(
                    L10n.string("ai.settings.apiKeyStored")
                        .replacingOccurrences(of: "{last4}", with: last4)
                )
                .font(Tokens.Typography.captionRelative)
                .foregroundStyle(Tokens.Palette.textSecondary.resolve(for: scheme))
            }

            Text(L10n.string("ai.settings.apiKeyWriteOnly"))
                .font(Tokens.Typography.captionRelative)
                .foregroundStyle(Tokens.Palette.textSecondary.resolve(for: scheme))
                .fixedSize(horizontal: false, vertical: true)
        }
    }

    private var pricing: some View {
        VStack(alignment: .leading, spacing: Tokens.Spacing.md) {
            SectionHeader(
                title: L10n.string("ai.settings.price"),
                subtitle: L10n.string("ai.settings.priceHint")
            )

            LabelledField(
                label: L10n.string("ai.inputPrice"),
                text: $inputPrice,
                isSecure: false,
                contentType: .plain,
                keyboard: .decimal
            )

            LabelledField(
                label: L10n.string("ai.outputPrice"),
                text: $outputPrice,
                isSecure: false,
                contentType: .plain,
                keyboard: .decimal
            )

            LabelledField(
                label: L10n.string("ai.settings.budget"),
                text: $budget,
                isSecure: false,
                contentType: .plain,
                keyboard: .decimal
            )
        }
    }

    /// The gate that decides whether clinical prompts run at all.
    @ViewBuilder
    private var retention: some View {
        if let settings = state.settings {
            VStack(alignment: .leading, spacing: Tokens.Spacing.md) {
                SectionHeader(title: L10n.string("ai.retention"))

                Card(tone: settings.zeroRetentionConfirmed ? .success : .warning) {
                    VStack(alignment: .leading, spacing: Tokens.Spacing.md) {
                        Toggle(
                            L10n.string("ai.settings.zeroRetention"),
                            isOn: Binding(
                                get: { settings.zeroRetentionConfirmed },
                                set: { confirmed in
                                    Task {
                                        await model.confirmRetention(confirmed, note: retentionNote)
                                        state = model.currentState()
                                    }
                                }
                            )
                        )
                        .font(Tokens.Typography.bodyRelative)
                        .frame(minHeight: Tokens.minimumTouchTarget)
                        .disabled(state.saving)

                        LabelledField(
                            label: L10n.string("ai.retentionNote"),
                            text: $retentionNote,
                            isSecure: false,
                            contentType: .plain,
                            keyboard: .default
                        )

                        if let note = settings.zeroRetentionNote, let at = settings.zeroRetentionAt {
                            FieldRow(
                                label: L10n.string("ai.retentionRecorded"),
                                value: "\(note) · \(at.formatted(date: .abbreviated, time: .omitted))"
                            )
                        }
                    }
                }
            }
        }
    }

    private var actions: some View {
        VStack(spacing: Tokens.Spacing.sm) {
            PrimaryButton(
                title: L10n.string("common.save"),
                isBusy: state.saving,
                isEnabled: !state.saving
            ) {
                _ = await model.save(
                    provider: provider,
                    model: chosenModel.isEmpty ? nil : chosenModel,
                    apiKey: apiKey,
                    inputPrice: inputPrice.isEmpty ? nil : inputPrice,
                    outputPrice: outputPrice.isEmpty ? nil : outputPrice,
                    budget: budget.isEmpty ? nil : budget
                )
                apiKey = ""
                state = model.currentState()
            }

            Button(L10n.string("ai.settings.test")) {
                Task {
                    await model.test()
                    state = model.currentState()
                }
            }
            .font(Tokens.Typography.calloutRelative)
            .frame(maxWidth: .infinity, minHeight: Tokens.minimumTouchTarget)
            .foregroundStyle(Tokens.Palette.accent.resolve(for: scheme))
            .disabled(state.saving)

            Button(L10n.string("ai.settings.clear")) {
                Task {
                    await model.switchOff()
                    state = model.currentState()
                }
            }
            .font(Tokens.Typography.calloutRelative)
            .frame(maxWidth: .infinity, minHeight: Tokens.minimumTouchTarget)
            .foregroundStyle(Tokens.Palette.critical.resolve(for: scheme))
            .disabled(state.saving)
        }
    }

    private func reload() async {
        await model.load()
        state = model.currentState()

        provider = state.settings?.provider
        chosenModel = state.settings?.model ?? ""
        inputPrice = state.settings?.inputPricePerMTok ?? ""
        outputPrice = state.settings?.outputPricePerMTok ?? ""
        budget = state.settings?.monthlyBudgetUsd ?? ""
    }
}

File: android/feature/aisettings-ui/src/main/kotlin/xyz/klinik/feature/aisettings/ui/AiSettingsScreen.kt
package xyz.klinik.feature.aisettings.ui

import androidx.compose.foundation.horizontalScroll
import androidx.compose.foundation.layout.Arrangement
import androidx.compose.foundation.layout.Box
import androidx.compose.foundation.layout.Column
import androidx.compose.foundation.layout.ColumnScope
import androidx.compose.foundation.layout.Row
import androidx.compose.foundation.layout.fillMaxSize
import androidx.compose.foundation.layout.fillMaxWidth
import androidx.compose.foundation.layout.heightIn
import androidx.compose.foundation.layout.padding
import androidx.compose.foundation.rememberScrollState
import androidx.compose.foundation.text.KeyboardOptions
import androidx.compose.foundation.verticalScroll
import androidx.compose.material3.Button
import androidx.compose.material3.Checkbox
import androidx.compose.material3.CircularProgressIndicator
import androidx.compose.material3.FilterChip
import androidx.compose.material3.OutlinedButton
import androidx.compose.material3.OutlinedTextField
import androidx.compose.material3.Surface
import androidx.compose.material3.Text
import androidx.compose.material3.TextButton
import androidx.compose.runtime.Composable
import androidx.compose.ui.Alignment
import androidx.compose.ui.Modifier
import androidx.compose.ui.semantics.heading
import androidx.compose.ui.semantics.semantics
import androidx.compose.ui.text.input.KeyboardType
import androidx.compose.ui.text.input.PasswordVisualTransformation
import androidx.compose.ui.text.style.TextAlign
import xyz.klinik.design.Tokens
import xyz.klinik.design.klinikColor
import xyz.klinik.feature.aisettings.AiSettingsDraft
import xyz.klinik.feature.aisettings.AiSettingsPhase
import xyz.klinik.feature.aisettings.AiSettingsState
import xyz.klinik.network.AiProviderChoice
import xyz.klinik.network.UiText

/** Text the screen needs, resolved by the caller from string resources. */
data class AiSettingsStrings(
    val title: String,
    val notPermitted: String,
    val retry: String,
    val provider: String,
    val model: String,
    val apiKey: String,
    val apiKeyWriteOnly: String,
    /** "Kayıtlı anahtar: •••• {last4}", with the four already in it. */
    val apiKeyStored: (String) -> String,
    val inputPrice: String,
    val outputPrice: String,
    val priceHint: String,
    val pricingPage: String,
    val budget: String,
    val retention: String,
    val zeroRetention: String,
    val zeroRetentionCleared: String,
    val retentionNote: String,
    val ready: String,
    val notReady: String,
    val notClinicalReady: String,
    val save: String,
    val test: String,
    /** "Bağlantı çalışıyor ({model})", with the version that answered in it. */
    val testOk: (String) -> String,
    val testFailed: String,
    val clear: String,
    val missingName: (String) -> String,
    val message: (UiText) -> String,
)

/**
 * Which model service the clinic uses, and on what terms (spec 3.4, 14.5).
 *
 * The key field is write-only and says so, because nothing can read it back —
 * what the screen shows instead is the last four characters, which answers the
 * only question anybody has about it.
 *
 * "Ready" and "ready for clinical work" are two different lines, deliberately.
 * Without the zero-retention declaration the layer still refuses every
 * clinical prompt, and a screen showing only `ready` would tell an
 * administrator they had finished while the interesting half was off.
 */
@Composable
fun AiSettingsScreen(
    state: AiSettingsState,
    strings: AiSettingsStrings,
    onChooseProvider: (AiProviderChoice) -> Unit,
    onEdit: (AiSettingsDraft.() -> AiSettingsDraft) -> Unit,
    onSave: () -> Unit,
    onTest: () -> Unit,
    onClear: () -> Unit,
    onOpenPricing: (String) -> Unit,
    onRetry: () -> Unit,
    modifier: Modifier = Modifier,
) {
    Surface(color = klinikColor("background"), modifier = modifier.fillMaxSize()) {
        when (val phase = state.phase) {
            AiSettingsPhase.Loading -> Centered { CircularProgressIndicator() }

            AiSettingsPhase.NotPermitted -> Centered {
                Text(
                    strings.notPermitted,
                    color = klinikColor("textSecondary"),
                    textAlign = TextAlign.Center,
                )
            }

            is AiSettingsPhase.Failed -> Centered {
                Column(horizontalAlignment = Alignment.CenterHorizontally) {
                    Text(strings.message(phase.message), color = klinikColor("critical"))
                    TextButton(
                        onClick = onRetry,
                        modifier = Modifier.heightIn(min = Tokens.minimumTouchTarget),
                    ) {
                        Text(strings.retry)
                    }
                }
            }

            AiSettingsPhase.Loaded -> Column(
                modifier = Modifier
                    .fillMaxSize()
                    .verticalScroll(rememberScrollState())
                    .padding(Tokens.Spacing.lg),
                verticalArrangement = Arrangement.spacedBy(Tokens.Spacing.lg),
            ) {
                Text(
                    strings.title,
                    fontSize = Tokens.Typography.heading.size,
                    fontWeight = Tokens.Typography.heading.weight,
                    color = klinikColor("textPrimary"),
                    modifier = Modifier.semantics { heading() },
                )

                Readiness(state, strings)

                state.error?.let {
                    Text(strings.message(it), color = klinikColor("critical"))
                }

                Section(strings.provider) {
                    Row(
                        modifier = Modifier.horizontalScroll(rememberScrollState()),
                        horizontalArrangement = Arrangement.spacedBy(Tokens.Spacing.sm),
                    ) {
                        state.providers.forEach { provider ->
                            FilterChip(
                                selected = provider.id == state.draft.provider,
                                onClick = { onChooseProvider(provider.id) },
                                label = { Text(provider.label) },
                                modifier = Modifier.heightIn(min = Tokens.minimumTouchTarget),
                            )
                        }
                    }

                    OutlinedTextField(
                        value = state.draft.model,
                        onValueChange = { value -> onEdit { copy(model = value) } },
                        label = { Text(strings.model) },
                        modifier = Modifier.fillMaxWidth(),
                    )
                }

                Section(strings.apiKey) {
                    state.settings?.apiKeyLast4?.let { last4 ->
                        Text(strings.apiKeyStored(last4), color = klinikColor("textSecondary"))
                    }

                    OutlinedTextField(
                        value = state.draft.apiKey,
                        onValueChange = { value -> onEdit { copy(apiKey = value) } },
                        label = { Text(strings.apiKey) },
                        visualTransformation = PasswordVisualTransformation(),
                        modifier = Modifier.fillMaxWidth(),
                    )

                    // Said before it is typed, not after it disappears.
                    Text(
                        strings.apiKeyWriteOnly,
                        fontSize = Tokens.Typography.caption.size,
                        color = klinikColor("textSecondary"),
                    )
                }

                Section(strings.priceHint) {
                    Price(strings.inputPrice, state.draft.inputPrice) { value ->
                        onEdit { copy(inputPrice = value) }
                    }
                    Price(strings.outputPrice, state.draft.outputPrice) { value ->
                        onEdit { copy(outputPrice = value) }
                    }
                    Price(strings.budget, state.draft.budget) { value ->
                        onEdit { copy(budget = value) }
                    }

                    state.chosenProvider?.let { provider ->
                        TextButton(
                            onClick = { onOpenPricing(provider.pricingUrl) },
                            modifier = Modifier.heightIn(min = Tokens.minimumTouchTarget),
                        ) {
                            Text(strings.pricingPage)
                        }
                    }
                }

                Section(strings.retention) {
                    // The provider's own terms, beside the box rather than
                    // behind a link: a tick against an unread sentence is not a
                    // record of anything.
                    state.chosenProvider?.let { provider ->
                        Text(provider.retentionNote, color = klinikColor("textSecondary"))
                    }

                    if (state.retentionNeedsConfirming) {
                        Text(strings.zeroRetentionCleared, color = klinikColor("warning"))
                    }

                    Row(
                        modifier = Modifier
                            .fillMaxWidth()
                            .heightIn(min = Tokens.minimumTouchTarget),
                        verticalAlignment = Alignment.CenterVertically,
                    ) {
                        Checkbox(
                            checked = state.draft.zeroRetentionConfirmed,
                            onCheckedChange = { on ->
                                onEdit { copy(zeroRetentionConfirmed = on) }
                            },
                        )
                        Text(strings.zeroRetention, color = klinikColor("textPrimary"))
                    }

                    state.settings?.zeroRetentionAt?.let { at ->
                        Text(
                            "${strings.retentionNote}: ${at.take(10)}",
                            fontSize = Tokens.Typography.caption.size,
                            color = klinikColor("textSecondary"),
                        )
                    }
                }

                state.testResult?.let { result ->
                    Text(
                        if (result.ok) {
                            strings.testOk(result.model.orEmpty())
                        } else {
                            "${strings.testFailed}${result.error?.let { ": $it" }.orEmpty()}"
                        },
                        color = if (result.ok) klinikColor("success") else klinikColor("critical"),
                    )
                }

                Button(
                    onClick = onSave,
                    enabled = state.canSave,
                    modifier = Modifier
                        .fillMaxWidth()
                        .heightIn(min = Tokens.minimumTouchTarget),
                ) {
                    Text(strings.save)
                }

                OutlinedButton(
                    onClick = onTest,
                    enabled = !state.busy && state.settings?.hasApiKey == true,
                    modifier = Modifier
                        .fillMaxWidth()
                        .heightIn(min = Tokens.minimumTouchTarget),
                ) {
                    Text(strings.test)
                }

                TextButton(
                    onClick = onClear,
                    enabled = !state.busy,
                    modifier = Modifier
                        .fillMaxWidth()
                        .heightIn(min = Tokens.minimumTouchTarget),
                ) {
                    Text(strings.clear)
                }
            }
        }
    }
}

/**
 * Running, and running for clinical work, are two statements.
 *
 * Without the declaration the layer refuses every clinical prompt, so a screen
 * that said only "ready" would tell an administrator they had finished while
 * the half the clinic cares about was still off.
 */
@Composable
private fun Readiness(state: AiSettingsState, strings: AiSettingsStrings) {
    val settings = state.settings ?: return

    Column(verticalArrangement = Arrangement.spacedBy(Tokens.Spacing.xs)) {
        Text(
            if (settings.ready) strings.ready else strings.notReady,
            color = if (settings.ready) klinikColor("success") else klinikColor("textSecondary"),
        )

        if (settings.ready && !settings.readyForClinicalUse) {
            Text(strings.notClinicalReady, color = klinikColor("warning"))
        }

        settings.missingStringKeys.forEach { key ->
            Text(
                strings.missingName(key),
                fontSize = Tokens.Typography.caption.size,
                color = klinikColor("warning"),
            )
        }
    }
}

@Composable
private fun Price(label: String, value: String, onChange: (String) -> Unit) {
    OutlinedTextField(
        value = value,
        onValueChange = onChange,
        label = { Text(label) },
        keyboardOptions = KeyboardOptions(keyboardType = KeyboardType.Decimal),
        modifier = Modifier.fillMaxWidth(),
    )
}

@Composable
private fun Section(title: String, content: @Composable ColumnScope.() -> Unit) {
    Surface(color = klinikColor("surface"), modifier = Modifier.fillMaxWidth()) {
        Column(
            modifier = Modifier.padding(Tokens.Spacing.lg),
            verticalArrangement = Arrangement.spacedBy(Tokens.Spacing.sm),
        ) {
            Text(
                title,
                fontSize = Tokens.Typography.subheading.size,
                fontWeight = Tokens.Typography.subheading.weight,
                color = klinikColor("textPrimary"),
                modifier = Modifier.semantics { heading() },
            )

            content()
        }
    }
}

@Composable
private fun Centered(content: @Composable () -> Unit) {
    Box(
        modifier = Modifier.fillMaxSize().padding(Tokens.Spacing.xl),
        contentAlignment = Alignment.Center,
    ) {
        Column(horizontalAlignment = Alignment.CenterHorizontally) { content() }
    }
}
